apps/omg_watcher/lib/omg_watcher/db/eth_event.ex

```diff
@@ -17,22 +17,27 @@ defmodule OMG.Watcher.DB.EthEvent do
   Ecto schema for events logged by Ethereum: deposits and exits
   """
   use Ecto.Schema
+  import Ecto.Changeset
 
   alias OMG.Crypto
   alias OMG.Utxo
   alias OMG.Watcher.DB
 
   require Utxo
 
-  @primary_key {:hash, :binary, []}
-  @derive {Phoenix.Param, key: :hash}
+  @primary_key {:rootchain_txhash, :binary, []}
```

Comment:
I don't think this is the case today (and hope I didn't miss this in our earlier discussions to lead you astray), but note that there may be multiple ethereum events fired at a single `rootchain_txhash`. I imagine `{rootchain_txhash, log_id}` would be the standard unique index, but this requires to be confirmed (c.f. `logIndex` in https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getfilterchanges)

Reply:
Interesting point. This design may still work, because it's a many-to-many with txoutputs. I deliberately considered the `process_exits` case. This affects many utxos and will create and DB mapping correctly.

On the other hand, this is one event affecting mutiple utxos, not multiple events fired from one ethereum txn.

Let me think about this a little more and look at your reference for `logIndex`. If it turns out we need to use `{rootchain_txhash, log_id}`, that'll be just a minor change. However, I'll unfortunately need to create another 'fake' column that will combine both values into one since ecto doesn't support many-to-many on composite columns.

Reply:
@pdobacz  Ok, I think this will work. `{rootchain_txhash, event_type}`. Then the many-to-many join table will join on a hash of these 2 values.

Thoughts?

Reply:
>  I think this will work. `{rootchain_txhash, event_type}`

Not necessarily, if we're putting on a "future-proofing" hat (which we are). It breaks if there are a couple of instances of the same `event_type` under the same `rootchain_txhash`. 

Oh in fact there's a simple counter example to my earlier assessment:

> I don't think this is the case today

`ExitFinalized` can be fired multiple times per root chain tx, so **it is the case already**.

Reply:
Ah! You're totally right. Thanks for pointing this out.

Reply:
Actually, can we hop on a call about this today or tomorrow?

I checked RootChain.sol and I don't see this with `ExitFinalized`. Further, every time `ExitFinalized` is fired, a new UTXO position is returned, thus the many-to-many relationship (ie: it's the same `ExitFinalized` event fired for multiple UTXOs. So I think my logic is currently still correct.

However, about future-proofing, you have a good point here and I'd just like to discuss more and understand your thoughts better.

Thanks!

Reply:
Yeah, it sort of depends on where you draw the line of what an `eth_event` is. If it's to have a 1-1 relationship with an ethereum event log (i.e. a single `emit ExitFinalized` executed), then you need `{root_chain_txhash, log_id}`.

If you prefer to "squash" multiple into a single row in this table, then `... event_type}` works. It sounds a bit unnatural to me though :thinking:.

Let's sync on slack & have a call!